glean-core/android/src/main/java/mozilla/telemetry/glean/private/LabeledMetricType.kt

```diff
@@ -17,134 +23,39 @@ package mozilla.telemetry.glean.private
-                seenLabels.addAll(
-                    it.getIdentifiersInStores((subMetric as CommonMetricData).sendInPings)
-                        .filter { it.startsWith(prefix) }
-                        .map { it.substring(prefix.length) }
-                )
-            }
-        }
-
-        if (!seenLabels.contains(label)) {
-            if (seenLabels.size >= MAX_LABELS) {
-                return OTHER_LABEL
-            } else {
-                if (label.length > MAX_LABEL_LENGTH) {
-                    recordError(
-                        this,
-                        ErrorType.InvalidValue,
-                        "label length ${label.length} exceeds maximum of $MAX_LABEL_LENGTH",
-                        logger
-                    )
-                    return OTHER_LABEL
-                }
-
-                // Labels must be snake_case.
-                if (!labelRegex.matches(label)) {
-                    recordError(
-                        this,
-                        ErrorType.InvalidValue,
-                        "label must be snake_case, got '$label'",
-                        logger
-                    )
-                    return OTHER_LABEL
-                }
-                seenLabels.add(label)
-            }
-        }
-        return label
-        */
-    }
-
-    /**
-     * Get a copy of the subMetric with the name changed to the given `newName`.
-     *
-     * @param newName The new name for the metric.
-     * @return A copy of subMetric with the new name.
-     * @throws IllegalStateException If this metric type does not support labels.
-     */
-    @Suppress("UNCHECKED_CAST")
-    internal fun getMetricWithNewName(newName: String): T {
-        // function is "internal" so we can mock it in testing
-
-        // Every metric that supports labels needs an entry here
-        // FIXME(bug 1552873): We need proper implementations AND serialization handling of these
-        return when (subMetric) {
-            is BooleanMetricType -> BooleanMetricType(disabled = disabled, category = category, name = newName, sendInPings = sendInPings, lifetime = lifetime) as T
-            is CounterMetricType -> CounterMetricType(disabled = disabled, category = category, name = newName, sendInPings = sendInPings, lifetime = lifetime) as T
-            /*is DatetimeMetricType -> subMetric.copy(name = newName) as T*/
-            /*is StringListMetricType -> subMetric.copy(name = newName) as T*/
-            is StringMetricType -> StringMetricType(disabled = disabled, category = category, name = newName, sendInPings = sendInPings, lifetime = lifetime) as T
-            /*is TimespanMetricType -> subMetric.copy(name = newName) as T*/
-            /*is UuidMetricType -> subMetric.copy(name = newName) as T*/
+    init {
+        val ffiPingsList = StringArray(sendInPings.toTypedArray(), "utf-8")
+        val labelList = if (labels != null) { StringArray(labels.toList().toTypedArray(), "utf-8") } else { null }
```

Comment:
nit: `val labelList = labels?.let { StringArray(...) }`

Reply:
Ha, I knew something existed but was unable to find it (ever noticed that Kotlin's thousand-and-one different operators are ungoogleable if you don't know their name in the first place?)

Reply:
Yes, I completely agree with you :\ The only reason I know is that we stumbled upon that earlier than you, so we bothered someone else :P